src/app/variables.ts

```diff
@@ -1,56 +1,87 @@
+import { Components } from '@types'
+
 /**
  * Helper to make sure that all envs are defined properly
  * @param name env variable name
  */
-const getEnv = (name: string): string => {
-  const value = process.env[`REACT_APP_${name}`]
+const getEnv = (name: string, defaultValue?: string): string => {
+  const value = process.env[`REACT_APP_${name}`] || defaultValue
   if (value) {
     return value
   }
   throw new Error(`Missing environment variable: REACT_APP_${name}`)
 }
 
 /**
- * API URL base
+ * Environment spec - production or development
+ */
+export const ENV = getEnv('ENV')
+
+/**
+ * Stable ID of the application
  */
-export const API_URL = getEnv('API_URL')
+export const ID = getEnv('ID')
 
 /**
  * App deploy URL base
  */
-const CONTEXT = getEnv('CONTEXT')
-const PRODUCTION_URL = getEnv('PRODUCTION_URL')
-const DEVELOPMENT_URL = getEnv('DEVELOPMENT_URL')
-const APP_URL = CONTEXT === 'production' ? PRODUCTION_URL : DEVELOPMENT_URL
+export const URL = (() => {
+  if (getEnv('NETLIFY', 'false') === 'false') {
+    // Not running on Netlify
+    return getEnv('URL')
+  } else if (getEnv('NETLIFY_CONTEXT') === 'production') {
+    // Main production instance on Netlify
+    return getEnv('NETLIFY_URL')
+  } else {
+    // Deploy preview or branch deploy on Netlify
+    return getEnv('NETLIFY_DEPLOY_PRIME_URL')
+  }
+})()
 
 /**
- * OIDC variables
+ * Components configuration
  */
-const OIDC_AUTHORITY = getEnv('OIDC_AUTHORITY')
-const OIDC_CLIENT_ID = getEnv('OIDC_CLIENT_ID')
+export const COMPONENTS: Components = (() => {
+  const base64String = getEnv('COMPONENTS')
```

Comment:
I figured that instead of dealing with error-prone quotes escaping within the JSON string, we could just encode everything using the base64 algo :-)